backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/textfield/internal/BpkTextFieldImpl.kt

```diff
@@ -240,65 +241,73 @@ private fun TextFieldBox(
             textFieldContent()
         }
 
-        if (trailingIcon != null) {
-            BpkIcon(
-                icon = trailingIcon,
-                contentDescription = null,
-                size = BpkIconSize.Small,
-                tint = animateColorAsState(
-                    when (status) {
-                        is BpkFieldStatus.Disabled -> BpkTheme.colors.textDisabled
-                        else -> BpkTheme.colors.textPrimary
-                    },
-                ).value,
-            )
-        } else {
-            var lastIcon by remember { mutableStateOf<Icon?>(null) }
-            when {
-                status is BpkFieldStatus.Validated -> lastIcon =
-                    Icon(icon = BpkIcon.TickCircle, size = BpkIconSize.Large, color = BpkTheme.colors.statusSuccessSpot)
-
-                status is BpkFieldStatus.Error -> lastIcon = Icon(
-                    icon = BpkIcon.ExclamationCircle,
-                    size = BpkIconSize.Large,
-                    color = BpkTheme.colors.statusDangerSpot,
-                )
+        TrailingIcon(trailingIcon, status, clearAction, value)
+    }
+}
 
-                status is BpkFieldStatus.Default && clearAction != null && value.text.isNotEmpty() -> lastIcon = Icon(
-                    icon = BpkIcon.CloseCircle,
-                    contentDescription = clearAction.contentDescription,
-                    size = BpkIconSize.Small,
-                    color = BpkTheme.colors.textSecondary,
-                    modifier = Modifier
-                        .clickable(bounded = false, role = Role.Button) {
-                            clearAction.onClick()
-                        }
-                        .testTag("textFieldClearButton"),
-                )
+@Composable
+private fun RowScope.TrailingIcon(
+    trailingIcon: BpkIcon?,
+    status: BpkFieldStatus,
+    clearAction: BpkClearAction?,
+    value: TextFieldValue,
+) {
+    var lastIcon by remember { mutableStateOf<Icon?>(null) }
+    val currentIcon = if (trailingIcon != null) {
```

Comment:
we need an easy way to know if the icon is visible - we keep the last icon around above for the visibility animation, however this allows us to know when an icon is actually visible and use that for the animation